### PR TITLE
Refactoring/feesharing to fee sharing collector

### PR DESCRIPTION
**Pull request by cwsnt**

### CODE REVIEW

File: contracts/governance/FeeSharingCollectorProxy/FeeSharingCollectorProxyStorage.sol [hunk 1]
```diff
@@ -3,18 +3,19 @@ pragma experimental ABIEncoderV2;
...
- * @notice Just the storage part of feeSharingProxy contract, no functions,
+ * @title FeeSharingCollectorProxy Storage contact.
+ * @notice Just the storage part of feeSharingCollectorProxy contract, no functions,
  * only constant, variables and required structures (mappings).
- * Used by FeeSharingProxy, and the implementation logic of FeeSharingProxy (FeeSharingLogic)
+ * Used by FeeSharingCollectorProxy, and the implementation logic of FeeSharingCollectorProxy (FeeSharingCollector)
  *
  * */
-contract FeeSharingProxyStorage is Ownable {
+contract FeeSharingCollectorProxyStorage is Ownable {
```
Comment (jameshowlett977) [inline contracts/governance/FeeSharingCollectorProxy/FeeSharingCollectorProxyStorage.sol:18]: FeeSharingCollectorStorage ?

File: contracts/governance/IFeeSharingCollectorProxy.sol [hunk 2]
```diff
@@ -1,10 +1,10 @@
 pragma solidity ^0.5.17;
 
 /**
- * @title Interface for contract governance/FeeSharingProxy/FeeSharingProxy.sol
+ * @title Interface for contract governance/FeeSharingCollectorProxy/FeeSharingCollectorProxy.sol
  * @dev Interfaces are used to cast a contract address into a callable instance.
  * */
-interface IFeeSharingProxy {
+interface IFeeSharingCollectorProxy {
```
Comment (jameshowlett977) [inline contracts/governance/IFeeSharingCollectorProxy.sol:7]: IFeeSharingCollector ?

**Review by jameshowlett977 — CHANGES_REQUESTED** [review]

File: contracts/governance/IFeeSharingCollectorProxy.sol (hunk 2 above)
Comment (cwsnt) [reply contracts/governance/IFeeSharingCollectorProxy.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/482/commits/e4758f4aa1e7376eafb188b06c4a85702f370842)

File: contracts/governance/Vesting/VestingRegistry.sol [hunk 3]
```diff
@@ -52,7 +52,7 @@ contract VestingRegistry is Ownable {
     address public staking;
 
     /// @notice Fee sharing proxy.
-    address public feeSharingProxy;
+    address public feeSharingCollectorProxy;
```
Comment (ororopickpocket) [inline contracts/governance/Vesting/VestingRegistry.sol:55]: same here

File: contracts/governance/Vesting/VestingStorage.sol [hunk 4]
```diff
@@ -24,7 +24,7 @@ contract VestingStorage is Ownable {
     address public tokenOwner;
 
     /// @notice Fee sharing Proxy.
-    IFeeSharingProxy public feeSharingProxy;
+    IFeeSharingCollector public feeSharingCollectorProxy;
```
Comment (ororopickpocket) [inline contracts/governance/Vesting/VestingStorage.sol:27]: same here. just calling it feeSharingCollector would be easier

File: contracts/governance/FeeSharingCollectorProxy/FeeSharingCollector.sol [hunk 5]
```diff
@@ -163,30 +162,28 @@ contract FeeSharingLogic is SafeMath96, IFeeSharingProxy, Ownable, FeeSharingPro
...
-                        "FeeSharingProxy::withdrawFees: pool token amount exceeds 96 bits"
+                        wrbtcAmountWithdrawn,
+                        "FeeSharingCollectorProxy::withdrawFeesAMM: wrbtc token amount exceeds 96 bits"
                     );
 
                 totalPoolTokenAmount = add96(
                     totalPoolTokenAmount,
                     amount96,
-                    "FeeSharingProxy::withdrawFees: total pool token amount exceeds 96 bits"
+                    "FeeSharingCollectorProxy::withdrawFeesAMM: total wrbtc token amount exceeds 96 bits"
```
Comment (ororopickpocket) [inline contracts/governance/FeeSharingCollectorProxy/FeeSharingCollector.sol:178]: also on all of these error messages it's strange to add "proxy".

File: contracts/governance/FeeSharingCollectorProxy/FeeSharingCollector.sol [hunk 6]
```diff
@@ -198,18 +195,42 @@ contract FeeSharingLogic is SafeMath96, IFeeSharingProxy, Ownable, FeeSharingPro
...
         /// @notice Transfer tokens from msg.sender
         bool success = IERC20(_token).transferFrom(address(msg.sender), address(this), _amount);
         require(success, "Staking::transferTokens: token transfer failed");
 
+        // if _token is wrbtc, need to unwrap it to rbtc
+        IWrbtcERC20 wrbtcToken = protocol.wrbtcToken();
+        if (_token == address(wrbtcToken)) {
+            wrbtcToken.withdraw(_amount);
+            _token = RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT;
+        }
```
Comment (ororopickpocket) [inline contracts/governance/FeeSharingCollectorProxy/FeeSharingCollector.sol:213]: these changes have nothing to do with the story and the title. a PR to a branch containing the fee sharing updates would be easier to review. could you merge the most recent dev branch into the base branch here?

File: contracts/governance/FeeSharingCollectorProxy/FeeSharingCollector.sol (hunk 5 above)
Comment (cwsnt) [reply contracts/governance/FeeSharingCollectorProxy/FeeSharingCollector.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/482/commits/61d88fbfe3ceb4b74fa8b612479c397faca5c290)

File: contracts/governance/Vesting/VestingStorage.sol (hunk 4 above)
Comment (cwsnt) [reply contracts/governance/Vesting/VestingStorage.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/482/commits/61d88fbfe3ceb4b74fa8b612479c397faca5c290)

File: contracts/governance/Vesting/VestingRegistry.sol (hunk 3 above)
Comment (cwsnt) [reply contracts/governance/Vesting/VestingRegistry.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/482/commits/61d88fbfe3ceb4b74fa8b612479c397faca5c290)

File: contracts/governance/Vesting/TeamVesting.sol [hunk 7]
```diff
@@ -35,13 +35,16 @@ contract TeamVesting is VestingStorage, Proxy {
         address _tokenOwner,
         uint256 _cliff,
         uint256 _duration,
-        address _feeSharingProxy
+        address _feeSharingCollectorProxy
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Vesting/TeamVesting.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/482/commits/61d88fbfe3ceb4b74fa8b612479c397faca5c290)

File: contracts/governance/FeeSharingCollectorProxy/FeeSharingCollector.sol (hunk 6 above)
Comment (cwsnt) [reply contracts/governance/FeeSharingCollectorProxy/FeeSharingCollector.sol]: merged the latest dev branch into this base branch

File: scripts/contractInteraction/contract_interaction.py [hunk 8]
```diff
@@ -51,6 +51,9 @@ def main():
     #sendFromMultisig(conf.contracts['Watcher'], 5e18)
 
     #withdrawRBTCFromIWRBTC('0x9BD6759F6D9eA15D33076e55d4CBba7cf85877A7', 1.6e18)
+    #sendMYNTFromMultisigToFeeSharingCollectorProxy(36632.144056847e18)
```
Comment (tjcloa) [inline scripts/contractInteraction/contract_interaction.py:54]:
```suggestion
    #sendMYNTFromMultisigToFeeSharingCollector(36632.144056847e18)
```

File: scripts/contractInteraction/token.py [hunk 9]
```diff
@@ -56,20 +56,20 @@ def increaseAllowanceFromMS(tokenContractAddress, receiver, amount):
     data = token.increaseAllowance.encode_input(receiver, amount)
     sendWithMultisig(conf.contracts['multisig'], tokenContractAddress, data, conf.acct)
 
-def sendMYNTFromMultisigToFeeSharingProxy(amount):
-    feeSharingProxy = Contract.from_abi("FeeSharingLogic", address=conf.contracts['FeeSharingProxy'], abi=FeeSharingLogic.abi, owner=conf.acct)
+def sendMYNTFromMultisigToFeeSharingCollectorProxy(amount):
```
Comment (tjcloa) [inline scripts/contractInteraction/token.py:59]:
```suggestion
def sendMYNTFromMultisigToFeeSharingCollector(amount):
```

**Review by tjcloa — CHANGES_REQUESTED** [review]
- StakingTest still has references to legacy FeeSharingProxy
- Folder FeeSharingCollectorProxy should be FeeSharingCollector
- Rename FeeSharingCollectorProxyTest.js to FeeSharingCollectorTest.js - we are not (only) testing the proxy functionality there
- Also see inline comments

**Review by tjcloa — CHANGES_REQUESTED** [review]
- StakingTest still has references to legacy FeeSharingProxy in comments
- Folder FeeSharingCollectorProxy should be FeeSharingCollector
- Rename FeeSharingCollectorProxyTest.js to FeeSharingCollectorTest.js - we are not (only) testing the proxy functionality there

**Review by tjcloa — APPROVED** [review]

**Pull request merged**